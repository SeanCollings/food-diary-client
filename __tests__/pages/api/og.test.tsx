import handler, { config } from '../../../pages/api/og';

jest.mock('@vercel/og', () => ({
  ImageResponse: jest.fn().mockImplementation((element: any) => element),
}));

describe('/api - og', () => {
  const mockError = jest.fn();

  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(mockError);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const mockUrl = 'https://mock.com';

  it('should return og image-response with no search-params', async () => {
    const actual = handler({
      method: 'GET',
      url: mockUrl,
    } as any);

    expect(actual).toMatchSnapshot();
  });

  it('should return og image-response with custom params', async () => {
    const actual = handler({
      method: 'GET',
      url: `${mockUrl}?title=a_new_title&path=a_new_path`,
    } as any);

    expect(actual).toMatchSnapshot();
  });

  it('should handle errors', async () => {
    try {
      handler({
        method: 'GET',
        url: '/',
      } as any);
    } catch (err) {
      expect(mockError).toHaveBeenCalled();
      expect((err as any).message).toMatchInlineSnapshot;
    }
  });

  it('should return config', () => {
    expect(config).toMatchInlineSnapshot(`
      {
        "runtime": "edge",
      }
    `);
  });
});
