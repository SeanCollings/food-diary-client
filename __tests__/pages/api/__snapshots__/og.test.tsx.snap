// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api - og should return og image-response with custom params 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "background": "#121212",
      "borderLeft": "2px solid #2B9348",
      "borderRight": "2px solid #2B9348",
      "display": "flex",
      "fontWeight": 400,
      "height": "100%",
      "justifyContent": "center",
      "letterSpacing": "-.02em",
      "width": "100%",
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "left": 42,
        "position": "absolute",
        "top": 42,
      }
    }
  >
    <span
      style={
        {
          "backgroundImage": "linear-gradient(to right,#55A630,#2B9348)",
          "borderRadius": "50%",
          "height": 24,
          "width": 24,
        }
      }
    />
    <span
      style={
        {
          "color": "#ffffffe6",
          "fontSize": 20,
          "marginLeft": 8,
        }
      }
    >
      foodandwellnessdiary.vercel.app//a_new_path
    </span>
  </div>
  <div
    style={
      {
        "backgroundColor": "#55A630",
        "borderRadius": "8px",
        "color": "#ffffffe6",
        "display": "flex",
        "flexWrap": "wrap",
        "fontSize": 40,
        "justifyContent": "center",
        "lineHeight": 1.4,
        "margin": "0 42px",
        "maxWidth": 550,
        "padding": "100px 50px",
        "textAlign": "center",
        "width": "auto",
      }
    }
  >
    a_new_title
  </div>
</div>
`;

exports[`/api - og should return og image-response with no search-params 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "background": "#121212",
      "borderLeft": "2px solid #2B9348",
      "borderRight": "2px solid #2B9348",
      "display": "flex",
      "fontWeight": 400,
      "height": "100%",
      "justifyContent": "center",
      "letterSpacing": "-.02em",
      "width": "100%",
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "left": 42,
        "position": "absolute",
        "top": 42,
      }
    }
  >
    <span
      style={
        {
          "backgroundImage": "linear-gradient(to right,#55A630,#2B9348)",
          "borderRadius": "50%",
          "height": 24,
          "width": 24,
        }
      }
    />
    <span
      style={
        {
          "color": "#ffffffe6",
          "fontSize": 20,
          "marginLeft": 8,
        }
      }
    >
      foodandwellnessdiary.vercel.app/
    </span>
  </div>
  <div
    style={
      {
        "backgroundColor": "#55A630",
        "borderRadius": "8px",
        "color": "#ffffffe6",
        "display": "flex",
        "flexWrap": "wrap",
        "fontSize": 40,
        "justifyContent": "center",
        "lineHeight": 1.4,
        "margin": "0 42px",
        "maxWidth": 550,
        "padding": "100px 50px",
        "textAlign": "center",
        "width": "auto",
      }
    }
  >
    One step at a time. Towards wellness.
  </div>
</div>
`;
